plugins/amd-gpu/fu-amd-gpu-atom-firmware.c

```diff
@@ -0,0 +1,424 @@
+			    "ATOMBIOS number of strings is 0");
+		return FALSE;
+	}
+	idx = fu_struct_atom_image_get_str_loc(atom_image);
+	if (idx == 0) {
+		g_set_error(error,
+			    G_IO_ERROR,
+			    G_IO_ERROR_FAILED,
+			    "ATOMBIOS string location is invalid");
+		return FALSE;
+	}
+
+	/* make sure there is enough space for all the strings */
+	atombios_size = fu_firmware_get_size(FU_FIRMWARE(self));
+	if ((gsize)(idx + (num_str * (STRLEN_NORMAL - 1))) > atombios_size) {
+		g_set_error(error,
+			    G_IO_ERROR,
+			    G_IO_ERROR_FAILED,
+			    "bufsz is too small for all strings");
+		return FALSE;
+	}
+
+	base = fu_firmware_get_addr(FU_FIRMWARE(self));
+
+	/* parse atombios strings */
+	for (i = 0; i < num_str; i++) {
+		len = strnlen((gchar *)(buf + base + idx), STRLEN_NORMAL - 1);
+		if (!fu_memcpy_safe((guint8 *)tmp,
+				    sizeof(tmp),
+				    0x0,
+				    buf,
+				    bufsz,
+				    base + idx,
+				    len,
+				    error)) {
+			g_prefix_error(error, "failed to copy string %d: ", i);
+			return FALSE;
+		}
+
+		switch (i) {
+		case FU_ATOM_STRING_INDEX_PART_NUMBER:
+			self->part_number = g_strndup((gchar *)tmp, len);
+			break;
+		case FU_ATOM_STRING_INDEX_ASIC:
+			self->asic = g_strndup((gchar *)tmp, len);
+			break;
+		case FU_ATOM_STRING_INDEX_PCI_TYPE:
+			self->pci_type = g_strndup((gchar *)tmp, len);
+			break;
+		case FU_ATOM_STRING_INDEX_MEMORY_TYPE:
+			self->memory_type = g_strndup(tmp, len);
+			break;
+		default:
+			g_set_error(error,
+				    G_IO_ERROR,
+				    G_IO_ERROR_FAILED,
+				    "unknown string index: %d",
+				    i);
+			return FALSE;
+		}
+		idx += len + 1;
+	}
+
+	/* make sure there is enough space for name string */
+	if ((gsize)(idx + 2 + STRLEN_LONG - 1) > atombios_size) {
+		g_set_error(error,
+			    G_IO_ERROR,
+			    G_IO_ERROR_FAILED,
+			    "bufsz is too small for name string");
+		return FALSE;
+	}
+
+	/* skip the following 2 chars: 0x0D 0x0A */
+	idx += 2;
+
+	/* get the model number */
+	len = strnlen((gchar *)buf + base + idx, STRLEN_LONG - 1);
+	g_debug("base 0x%lx, idx 0x%lx, len 0x%lx", base, idx, len);
+	if (!fu_memcpy_safe((guint8 *)tmp, sizeof(tmp), 0x0, buf, bufsz, base + idx, len, error)) {
+		g_prefix_error(error, "failed to copy model string: ");
+		return FALSE;
+	}
+	self->model_name = fu_strstrip((gchar *)tmp);
+
+	return TRUE;
+}
+
+static gboolean
+fu_amd_gpu_atom_parse_config_filename(FuAmdGpuAtomFirmware *self,
+				      const guint8 *buf,
+				      gsize bufsz,
+				      GByteArray *atom_header,
+				      GError **error)
+{
+	gchar tmp[STRLEN_LONG];
+	guint64 base;
+	guint16 idx;
+	gsize len;
+
+	base = fu_firmware_get_addr(FU_FIRMWARE(self));
+	idx = fu_struct_atom_rom21_header_get_config_filename_offset(atom_header);
+	len = strnlen((gchar *)(buf + base + idx), STRLEN_NORMAL - 1);
+	if (!fu_memcpy_safe((guint8 *)tmp, sizeof(tmp), 0x0, buf, bufsz, base + idx, len, error)) {
+		g_prefix_error(error, "failed to copy config string: ");
+		return FALSE;
+	}
```

Comment:
fu_memstrsafe, but maybe you want to wrap it in a static function that includes the strstrip